pkg/cloud/awsprovider.go

```diff
@@ -1896,6 +1899,7 @@ func parseSpotData(bucket string, prefix string, projectID string, region string
 		klog.V(5).Infof("ListObjects \"s3://%s/%s\" produced no keys", *ls2.Bucket, *ls2.Prefix)
 	}
 
+	// TODO: Worth it to use LastModifiedDate to determine if we should reparse the spot data?
```

Comment:
We'd need to keep track of a LastModifiedDate in the AWS struct and check against that for each file in the set. We could avoid downloading and parsing the files in this case, which may let us be more aggressive in our refresh times. Either way, we're still subject to the delay from AWS -> S3